sdk/resourcemanager/azure-resourcemanager-containerregistry/src/main/java/com/azure/resourcemanager/containerregistry/implementation/RegistryImpl.java

```diff
@@ -226,8 +226,8 @@ public boolean isDedicatedDataEndpointsEnabled() {
     }
 
     @Override
-    public ZoneRedundancy zoneRedundancy() {
-        return this.innerModel().zoneRedundancy();
+    public boolean isZoneRedundancyEnabled() {
+        return !Objects.isNull(this.innerModel().zoneRedundancy()) && ZoneRedundancy.ENABLED.equals(this.innerModel().zoneRedundancy());
```

Comment:
ResourceManagerUtils.toPrimitiveBoolean

Reply:
The input parameter of this static method `ResourceManagerUtils.toPrimitiveBoolean` must be of type `Boolean`, but the type of `ZoneRedundancy` is an enumeration. So I think `ResourceManagerUtils.toPrimitiveBoolean` is not applicable here.

Reply:
Seems the logic can be simplified to just `ZoneRedundancy.ENABLED.equals(this.innerModel().zoneRedundancy()`?

Reply:
The judgment of `isNull` is still needed.
![image](https://github.com/Azure/azure-sdk-for-java/assets/74638143/b28e5de3-1fc5-4d75-a9f0-795491583639)

Reply:
When `zoneRedundancy` is null, it means `isZoneRedundancyEnabled` is false, right? Seems to align with `ZoneRedundancy.ENABLED.equals(this.innerModel().zoneRedundancy()`'s result.
Anyway, no hurt to have the null check.

Reply:
Fixed in the [pr#38592](https://github.com/Azure/azure-sdk-for-java/pull/38592)